src/app/child-dev-project/children/children.service.ts

```diff
@@ -23,12 +24,14 @@ export class ChildrenService {
     private entitySchemaService: EntitySchemaService,
     private dbIndexing: DatabaseIndexingService,
     childPhotoService: ChildPhotoService,
-    private logger: LoggingService
+    private logger: LoggingService,
+    private entityConfigService: EntityConfigService
   ) {
     this.entitySchemaService.registerSchemaDatatype(
       new LoadChildPhotoEntitySchemaDatatype(childPhotoService)
     );
     this.createDatabaseIndices();
+    this.entityConfigService.addConfigAttributes<Child>(Child);
```

Comment:
A point for discussion is, where this line should be inserted. There are a few options:

- In the constructor of the `EntityMapperService`. This would ensure that all entities are registered when they are needed. But it will also register entities that might not be needed at all and it creates dependencies from the  `EntityMapperService` towards all entities and the `EntityConfigService`.
- In the load and save functions of the `EntityMapperService` this would ensure that an entity is only registered once it is actually needed. A list of registered entities would ensure that each entity is only registered once. This appraoch would also introduce the dependency towards the `EntityConfigService` and I didn't yet find a way to get the class (e.g. `Child`) when I only have an object of this class (e.g. `new Child("...")`). This is the case in the `load` function of the `EntityMapperService`. However, the class is needed in order to set up the `schema` of this class.
- In the `app.module.ts` constructor. This would also mean that all classes are registered no matter if they are used or not.

Reply:
But modules can load this class without using the module (through the `EntitiyMapperServic`). If this happens before the `ChildrenModule` was used, it would result in the `Child` class not being configured.

Reply:
I now even moved it to `app.module` to ensure that no entity can be loaded or saved before it has been registered. 
To make it more "lazy", it could be moved to the `EntityMapper` but this would introduce new dependencies.